src/Client/Connection.cpp

```diff
@@ -833,9 +857,23 @@ void Connection::sendQuery(
     /// Per query settings.
     if (settings)
     {
+        std::optional<Settings> modified_settings;
+        const Settings * settings_to_send = settings;
+        if (!settings_from_server.empty())
+        {
+            /// Don't send settings that we got from the server in the first place.
+            modified_settings.emplace(*settings);
+            for (const SettingChange & change : settings_from_server)
+            {
+                if (settings->get(change.name) == change.value)
+                    modified_settings->setDefaultValue(change.name);
```

Comment:
To me it seems that if `settings->get(change.name) == change.value` condition is true, then we should remove change from `modified_settings` in this case, why set the default value instead? AFAIU it then will be sent to server, but it would be incorrect value.